server/src/test/java/org/elasticsearch/index/mapper/NumberFieldMapperTests.java

```diff
@@ -70,6 +70,21 @@ protected void registerParameters(ParameterChecker checker) throws IOException {
                 },
                 m -> assertThat((m).onScriptError, equalTo("continue")));
         }
+
+        // dimension can be set from false to true, but not vice versa
+        checker.registerUpdateCheck(b -> b.field("dimension", true),
```

Comment:
I think the last time we talked we settled on making it fully immutable - no one way transitions. Is that right, @imotov ?

Reply:
Yes, that's what we discussed. I am still working on this. The PR is WIP

Reply:
Indeed that's the conclusion that we reached. 

My understanding that we are moving away from WIP label towards Draft PRs. I found draft PRs really in your face in terms of distinguishing between WIP and Ready for review PRs 😄  so I typically open all my WIP PRs as Drafts.

Reply:
Converted this to draft PR and removed the `WIP` label. Thanks Igor